### PR TITLE
Set up the structure for testing and for retrieving lists. Not comple…

### PR DESCRIPTION
**Pull request by rynkwn**
…tely secure.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def create
         if employee && employee.authenticate(params[:password])
           employee.auth = SecureRandom.urlsafe_base64 32
           employee.auth_expires_at = Time.now.to_i + 19600
-          render :json => {auth: employee.auth, auth_expires_at: employee.auth_expires_at}
+          render :json => {id: employee.id, auth: employee.auth, auth_expires_at: employee.auth_expires_at}
```
Comment (BenDrews) [inline app/controllers/sessions_controller.rb]: In my branch I have id: as employee: , it's a bit more explicit. Up to you which we go with.

Comment (rynkwn) [inline app/controllers/sessions_controller.rb]: Sure, I've made the change.

File: app/controllers/qr_codes_controller.rb [hunk 2]
```diff
@@ -29,10 +29,19 @@ def show
     end
   end
   
+  def listings
+    if employee = Employee.find_by(id: params[:employee_id])
+      #if employee.auth == params[:auth] && employee.auth_expires_at >= Time.current 
+        # Check to make sure that authorization token is current, and that the token isn't expired.
+        qr_codes = QrCode.where("company_id = " + employee.employer_id.to_s)
```
Comment (kevinmichaelchen) [inline app/controllers/qr_codes_controller.rb:36]:
``` ruby
QrCode.where(company_id: employee.employer_id.to_s)
```

File: db/seeds.rb [hunk 3]
```diff
@@ -1,2 +1,8 @@
 Employer.create(name: 'Urtu', username: 'urtu', password: 'phoenix', company_name: 'Urtu Inc.')
-Employee.create(name: 'Ryan Kwon', username: 'rhk1', password: 'phoenix', employer_id: 1)
\ No newline at end of file
+Employee.create(name: 'Ryan Kwon', username: 'rhk1', password: 'phoenix', employer_id: 1)
+QrCode.create(title: "Stuff1", body: "Hey", company_id: "1")
```
Comment (kevinmichaelchen) [inline db/seeds.rb:3]: I recommend the `create!` method over `create` since the former will throw an exception.

Comment (rynkwn) [inline db/seeds.rb:3]: Made the edit

File: app/controllers/qr_codes_controller.rb (hunk 2 above)
Comment (rynkwn) [inline app/controllers/qr_codes_controller.rb:36]: I agree. I found a stack overflow question that advocated raw SQL statements, but I think the use-case was pretty noticeably different from mine.

**Pull request merged**